streams/src/test/java/org/apache/kafka/streams/tests/BrokerCompatibilityTest.java

```diff
@@ -70,17 +74,27 @@ public static void main(final String[] args) {
         streamsProperties.put(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG, timeout + 1);
         //TODO remove this config or set to smaller value when KIP-91 is merged
         streamsProperties.put(StreamsConfig.producerPrefix(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG), 60000);
+        Serde<String> stringSerde = Serdes.String();
 
 
         final StreamsBuilder builder = new StreamsBuilder();
-        builder.stream(SOURCE_TOPIC).to(SINK_TOPIC);
+        builder.<String, String>stream(SOURCE_TOPIC).groupByKey(Serialized.with(stringSerde, stringSerde))
+            .count()
```

Comment:
Curious why this logic would make it fail fast?

Reply:
Without the explicit broker version check, the topology in its current form runs successfully on the `0.10.0.0` broker test. We need to add a stateful operation (can't create an internal topic) to fail fast for the `0.10.0.0` broker check.  IMHO supporting stateless operations on a `0.10.0.0` broker is ok.  WDYT?

Reply:
Could you elaborate a bit more? The new topology still do not have any internal topic, only adding a state store. Why it can trigger the failure while the previous simple pipe topology can not?